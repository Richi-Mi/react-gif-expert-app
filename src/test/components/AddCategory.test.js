import '@testing-library/jest-dom'

import React from 'react'
import AddCategory from "../../components/AddCategory"
import { shallow } from "enzyme"

describe(' 3.- Pruebas en <AddCategory /> ', () => {
    const setCategories = jest.fn(); // Es una función para que podamos sabder si se llamo o no
    let wrapper = shallow( <AddCategory setCategories={ setCategories }/> );
    
    beforeEach( () => {
        jest.clearAllMocks(); // Limpia todas las simulaciones
        wrapper = shallow( <AddCategory setCategories={ setCategories }/> );
    } );

    test(' * Debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    test(' * Debe de Cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        
        // elemento.simulate('evento', argumentos[e]) .- Simula eventos en las pruebas
        input.simulate('change', { target: { value } });
        const txt_p = wrapper.find('p').text().trim();

        expect( txt_p ).toBe( value );

    })
    test(' * Si las condiciones no se cumplen, no debe de hacer el posteo de información', () => {
        wrapper.find('form').simulate('submit', { preventDefault: () => { } });

        expect( setCategories ).not.toHaveBeenCalled() // Sirve para saber si la función fue llamada o no
    })
    test(' * Debe de llamar el setCategories, y limpiar la Caja de texto', () => {
        // 1.- Simular el inputChange
        const input = wrapper.find('input');
        const value = 'One Punch Man';

        input.simulate('change', { target: { value } } );
        // 2.- Simular el Submit
        const form = wrapper.find('form');
        form.simulate('submit', { preventDefault: () => {} });
        // 3.- setCategories, debio de haberse llamado
        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1); // Solo tiene que llamarse una vez la función
        expect( setCategories ).toHaveBeenCalledWith( expect.any( Function )); // La función tiene que llamarse con un argumento que sea una función.
        // 4.- el valor del input debe ser ''
        expect( input.prop('value') ).toBe('');
    })
    
      
})