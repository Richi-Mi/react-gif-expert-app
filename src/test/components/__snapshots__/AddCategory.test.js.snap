// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 3.- Pruebas en <AddCategory />   * Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    placeholder="Type the Category"
    type="text"
    value=""
  />
</form>
`;
